src/LibObjectFile/Elf/ElfObjectFile.cs

```diff
@@ -584,20 +584,22 @@ public bool TryWrite(Stream stream, out DiagnosticBag diagnostics)
         /// <returns><c>true</c> if the stream contains an ELF file. otherwise returns <c>false</c></returns>
         public static bool IsElf(Stream stream)
         {
-            return IsElf(stream, out _);
+            return IsElf(stream, out _, out _);
         }
 
         /// <summary>
         /// Checks if a stream contains an ELF file by checking the magic signature.
         /// </summary>
         /// <param name="stream">The stream containing potentially an ELF file</param>
         /// <param name="encoding">Output the encoding if ELF is <c>true</c>.</param>
+        /// <param name="class">Output the file class if ELF is <c>true</c>.</param>
         /// <returns><c>true</c> if the stream contains an ELF file. otherwise returns <c>false</c></returns>
-        public static bool IsElf(Stream stream, out ElfEncoding encoding)
+        public static bool IsElf(Stream stream, out ElfEncoding encoding, out ElfFileClass @class)
```

Comment:
Not sure about this one. 

Actually, this was an oversight, the code is missing a call to `ElfObjectFileExtensions.CopyIndentFrom`.

It would be probably better to let IsElf as it is, but to implement a private method like `ElfObjectFile ReadElfObjectFileSignature()` that does a bit of IsElf but return an ElfObjectFile and would then be used by TryRead.
ReadElfObjectFileSignature should use `ElfObjectFileExtensions.CopyIndentFrom`